src/chronify/utils/path_utils.py

```diff
@@ -0,0 +1,34 @@
+import shutil
+
+from pathlib import Path
+
+from chronify.exceptions import InvalidOperation
+
+
+def check_overwrite(path: Path, overwrite: bool) -> None:
+    """Check if the path exists, handling the user flag overwrite."""
+    if not path.exists():
+        return
+
+    if overwrite:
+        if path.is_dir():
+            shutil.rmtree(path)
+        else:
+            path.unlink()
+    else:
+        msg = f"{path=} already exists. Choose a different path or set overwrite=True."
+        raise InvalidOperation(msg)
+
+
+def delete_if_exists(path: Path) -> None:
+    """Delete the path if it exists, handling files and directories."""
+    if path.exists():
+        if path.is_dir():
+            shutil.rmtree(path)
+        else:
+            path.unlink()
+
+
+def to_path(path: Path | str) -> Path:
+    """Convert the instance to a Path if is not already one."""
+    return path if isinstance(path, Path) else Path(path)
```

Comment:
I can't remember your rationale for this, but why can't we just do `Path(path)` for either cases again?

Reply:
I did feel a bit silly writing this. The reason is that the constructor for Path runs a lot of logic instead of simply returning the existing object. The _significant_ speed difference doesn't matter for us, and so I could remove it if it is distracting.
```
In [1]: from pathlib import Path

In [2]: def to_path(path: Path | str) -> Path:
   ...:      return path if isinstance(path, Path) else Path(path)
   ...: 

In [3]: path1 = Path("dir1/dir2/dir3/dir4/dir5")

In [4]: %timeit to_path(path1)
30 ns ± 0.343 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)

In [5]: %timeit Path(path1)
1.17 μs ± 6.85 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

Reply:
Add a comment on why.